torch/autograd/profiler.py

```diff
@@ -1148,9 +1149,9 @@ def parse_kineto_results(result):
             for mem_record in mem_records:
                 if (mem_record[0].start_us() >= kineto_event.start_us() and
                         mem_record[0].start_us() <= abs_end_us):
```

Comment:
this piece of code has performance penalty as we tested. 

https://github.com/pytorch/pytorch/issues/58983 
https://github.com/pytorch/pytorch/issues/61041
https://github.com/pytorch/kineto/issues/308 are complaining this performance issue when handling memory events.

https://github.com/pytorch/pytorch/pull/60432 improve the performance of memory paring

Reply:
I'll rebase after your fix lands